llvm/include/llvm/Support/ModRef.h

```diff
@@ -56,6 +56,11 @@ enum class ModRefInfo : uint8_t {
 /// Debug print ModRefInfo.
 LLVM_ABI raw_ostream &operator<<(raw_ostream &OS, ModRefInfo MR);
 
+enum class InaccessibleTargetMemLocation {
```

Comment:
Each Target should have an InaccessibleTargetMemLocation, but I am not sure if we should check which Target it is being used in Support, Attributes and so on. I  would appreciate some suggestion if this is the right way to do this.

And this is still using Data to model the Memory Locations with ModRef. Is this the only way to add new memory location?

Reply:
Hi @antoniofrighetto,
I think your suggestion need me to change the way Memory Location is implemented.
However I did follow @nikic's solution(or I think I did) and create a generic name for the target memory locations, and they are printed according to Triple.  I think this makes It generic enough. 
Let me know what you think about the latest change. 
I have some follow up patches :
https://github.com/llvm/llvm-project/pull/154141
https://github.com/llvm/llvm-project/pull/154144
https://github.com/llvm/llvm-project/pull/155590
that I will rebase once this patch is merged.